STM32/Filtering/Src/array-math.c

```diff
@@ -123,6 +125,48 @@ double maMean(moving_avg_cbuf_handle_t p_ma, double newVal)
     return p_ma->sum / p_ma->cbuf_t.len;
 }
 
+/*!
+** @brief Computes the variance of an array using Welford's online algorithm.
+** @note The variance algorithm is equivalent to 
+**              var (X) = (1 / (N-1)) * SUM_i ((X(i) - mean(X))^2)
+**       Hence, it uses the unbiased (N-1) population variance, rather than the sample variance used in the references
+**       below. This will yield a more conservative estimate especially for smaller sample size.
+**
+**       Information about the algorithm can be found here: 
+**              Theory: https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm
+**              Implementation: https://stackoverflow.com/a/6664212
+*/
+double maVariance(moving_avg_cbuf_handle_t p_ma, double newVal)
+{
+    // Keep track of the previous average
+    static double avg = 0;
+    static double x_prev = 0;
```

Comment:
These statics should be part of the p_ma handle. It won't work for multi-handle implementation

Reply:
**avg** could be collected from p_ma.sum/p_ma.cbuf_t.len instead of static. Adds a divide, which TBF you may not want.

**x_prev** we should really have cb_getHead() or something that returns this